{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run ALF simulations\n",
    "\n",
    "- two datasets\n",
    "- same number of species as in Jeremy's empirical dataset\n",
    "\n",
    "### Other parameters currently:\n",
    "\n",
    "unitIsPam := true:\n",
    "\n",
    "#### parameters concerning the root genome\n",
    "realseed := false;\n",
    "protStart := 200;\n",
    "minGeneLength := 50;\n",
    "gammaLengthDist := [3, 133.8063];\n",
    "blocksize := 1:\n",
    "\n",
    "#### parameters concerning the species tree\n",
    "treeType := 'BDTree';\n",
    "birthRate := 0.01;\n",
    "deathRate := 0.001;\n",
    "NSpecies := 25;\n",
    "ultrametric := false;\n",
    "mutRate := 250;\n",
    "scaleTree := true;\n",
    "\n",
    "#### parameters concerning the substitution models\n",
    "substModels := [SubstitutionModel('LG')];\n",
    "indelModels := [IndelModel(0.0001, ZIPF, [1.821], 50)];\n",
    "rateVarModels := [RateVarModel()];\n",
    "modelAssignments := [1]:\n",
    "modelSwitchS := [[1]]:\n",
    "modelSwitchD := [[1]]:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: /usr/local/bin/../lib/simulator/splashscreen.txt: No such file or directory\n",
      "Darwin: Sequence Searching Facility\n",
      "Version 4.0, 2014-02-21\n",
      "  (c) E.T.H. Zurich\n",
      "> paramfile := 'alf-params_lopho.drw': wdir := './': ReadProgram(libname.'/simulator/evolstart.drw'): done:\n",
      "\n",
      "IO procedures for Synthetic Evolution sucessfully loaded\n",
      "Procedures for Genome Rearrangement sucessfully loaded\n",
      "Procedures for Synthetic Evolution sucessfully loaded\n",
      "\n",
      "\n",
      "parameter file read\n",
      "setting up work directory\n",
      "determining tree\n",
      "\n",
      "tree loaded\n",
      "model assignments done\n",
      "\n",
      "\n",
      "Evolution!\n",
      "\n",
      "\n",
      "species 2 was created at time 0.0000\n",
      "species 3 was created at time 1.3858\n",
      "species 4 was created at time 16.6214\n",
      "species 5 was created at time 27.5709\n",
      "species 6 was created at time 28.4886\n",
      "species 7 was created at time 30.2859\n",
      "species 8 was created at time 44.5080\n",
      "species 9 was created at time 48.0943\n",
      "species 10 was created at time 48.4042\n",
      "species 11 was created at time 49.4870\n",
      "species 12 was created at time 62.9200\n",
      "species 13 was created at time 68.6194\n",
      "species 14 was created at time 70.2564\n",
      "species 15 was created at time 71.1009\n",
      "species 16 was created at time 74.3100\n",
      "species 17 was created at time 75.7006\n",
      "species 18 was created at time 79.3465\n",
      "species 19 was created at time 86.7478\n",
      "species 20 was created at time 91.0164\n",
      "species 21 was created at time 91.6835\n",
      "species 22 was created at time 95.9135\n",
      "species 23 was created at time 95.9362\n",
      "species 24 was created at time 96.3155\n",
      "species 25 was created at time 106.9774\n",
      "species 26 was created at time 107.6954\n",
      "species 27 was created at time 110.4896\n",
      "species 28 was created at time 110.8539\n",
      "species 29 was created at time 112.3608\n",
      "species 30 was created at time 112.4415\n",
      "species 31 was created at time 114.1185\n",
      "species 32 was created at time 117.1256\n",
      "species 33 was created at time 122.0133\n",
      "species 34 was created at time 124.5154\n",
      "species 35 was created at time 124.5551\n",
      "species 36 was created at time 127.2748\n",
      "species 37 was created at time 130.0334\n",
      "species 38 was created at time 130.3008\n",
      "species 39 was created at time 131.1156\n",
      "species 40 was created at time 133.5454\n",
      "species 41 was created at time 133.6154\n",
      "species 42 was created at time 133.8439\n",
      "species 43 was created at time 134.7742\n",
      "species 44 was created at time 136.9642\n",
      "species 45 was created at time 141.8442\n",
      "species 46 was created at time 142.6227\n",
      "species 47 was created at time 143.4781\n",
      "species 48 was created at time 144.2980\n",
      "species 49 was created at time 147.1416\n",
      "species 50 was created at time 148.5041\n",
      "species 51 was created at time 149.4407\n",
      "species 52 was created at time 162.3400\n",
      "species 53 was created at time 162.8987\n",
      "species 54 was created at time 165.9160\n",
      "species 55 was created at time 167.7401\n",
      "species 56 was created at time 172.4657\n",
      "species 57 was created at time 178.6936\n",
      "species 58 was created at time 184.2282\n",
      "species 59 was created at time 186.1518\n",
      "species 60 was created at time 199.1406\n",
      "species 61 was created at time 200.8513\n",
      "species 62 was created at time 208.9790\n",
      "...real genome tree saved.\n",
      "...main calculation finished after 0.44 minutes.\n",
      "tar: Option --remove-files is not supported\n",
      "Usage:\n",
      "  List:    tar -tf <archive-filename>\n",
      "  Extract: tar -xf <archive-filename>\n",
      "  Create:  tar -cf <archive-filename> [filenames...]\n",
      "  Help:    tar --help\n",
      "...pairwise relationships written.\n",
      "...gene trees written.\n",
      "...real MSAs written.\n",
      "\n",
      "\n",
      ".db files written...\n",
      "\n",
      "\n",
      "Reading 175978 characters from file ./results_lopho//sim_match_lophotroch/DB/SE001.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE001.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE001.db.tree with 81775 entries\n",
      "Pat index with 81775 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVQKYQLTGQQVAFSPL\"\n",
      "Reading 179869 characters from file ./results_lopho//sim_match_lophotroch/DB/SE002.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE002.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE002.db.tree with 82070 entries\n",
      "Pat index with 82070 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYLADIIFGKGKIGQAL\"\n",
      "Reading 205623 characters from file ./results_lopho//sim_match_lophotroch/DB/SE003.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE003.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE003.db.tree with 83304 entries\n",
      "Pat index with 83304 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYWTPVNAKTITVPKEYKL\"\n",
      "Reading 193657 characters from file ./results_lopho//sim_match_lophotroch/DB/SE004.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE004.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE004.db.tree with 82913 entries\n",
      "Pat index with 82913 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSVISQVVIDRIQGRLH\"\n",
      "Reading 199352 characters from file ./results_lopho//sim_match_lophotroch/DB/SE005.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE005.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE005.db.tree with 83084 entries\n",
      "Pat index with 83084 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLLHLKKYRAKSYMEYV\"\n",
      "Reading 182888 characters from file ./results_lopho//sim_match_lophotroch/DB/SE006.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE006.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE006.db.tree with 82789 entries\n",
      "Pat index with 82789 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTYVQATASNNDQTYCD\"\n",
      "Reading 223676 characters from file ./results_lopho//sim_match_lophotroch/DB/SE007.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE007.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE007.db.tree with 82738 entries\n",
      "Pat index with 82738 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYKEGMVKLWVGGGYETT\"\n",
      "Reading 211455 characters from file ./results_lopho//sim_match_lophotroch/DB/SE008.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE008.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE008.db.tree with 82604 entries\n",
      "Pat index with 82604 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYVNGEAFKQGALAISLLL\"\n",
      "Reading 197664 characters from file ./results_lopho//sim_match_lophotroch/DB/SE009.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE009.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE009.db.tree with 82694 entries\n",
      "Pat index with 82694 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLGIYIEEGKKVMTGAA\"\n",
      "Reading 189979 characters from file ./results_lopho//sim_match_lophotroch/DB/SE010.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE010.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE010.db.tree with 82881 entries\n",
      "Pat index with 82881 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLYEFDYEGRFSPGTSL\"\n",
      "Reading 180933 characters from file ./results_lopho//sim_match_lophotroch/DB/SE011.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE011.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE011.db.tree with 82745 entries\n",
      "Pat index with 82745 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYPFHTKRGLRATYFKDY\"\n",
      "Reading 175961 characters from file ./results_lopho//sim_match_lophotroch/DB/SE012.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE012.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE012.db.tree with 82670 entries\n",
      "Pat index with 82670 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVQAVTSTMVSRDQSMQ\"\n",
      "Reading 183832 characters from file ./results_lopho//sim_match_lophotroch/DB/SE013.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE013.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE013.db.tree with 83081 entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pat index with 83081 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYPLSLEKSLVLRNRKVY\"\n",
      "Reading 175487 characters from file ./results_lopho//sim_match_lophotroch/DB/SE014.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE014.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE014.db.tree with 83073 entries\n",
      "Pat index with 83073 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYGLDPVRDVMQGIISFG\"\n",
      "Reading 200296 characters from file ./results_lopho//sim_match_lophotroch/DB/SE015.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE015.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE015.db.tree with 83288 entries\n",
      "Pat index with 83288 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSRYFSSEDVLSADPQN\"\n",
      "Reading 182401 characters from file ./results_lopho//sim_match_lophotroch/DB/SE016.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE016.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE016.db.tree with 83140 entries\n",
      "Pat index with 83140 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVLWIISVENILGNWLH\"\n",
      "Reading 189091 characters from file ./results_lopho//sim_match_lophotroch/DB/SE017.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE017.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE017.db.tree with 82735 entries\n",
      "Pat index with 82735 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVLYLDPIRIVGGAKNK\"\n",
      "Reading 199126 characters from file ./results_lopho//sim_match_lophotroch/DB/SE018.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE018.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE018.db.tree with 82877 entries\n",
      "Pat index with 82877 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVIDYDCYTKWADFISL\"\n",
      "Reading 206382 characters from file ./results_lopho//sim_match_lophotroch/DB/SE019.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE019.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE019.db.tree with 82907 entries\n",
      "Pat index with 82907 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYNLYIKREDGIYSYRGN\"\n",
      "Reading 198679 characters from file ./results_lopho//sim_match_lophotroch/DB/SE020.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE020.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE020.db.tree with 83229 entries\n",
      "Pat index with 83229 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSGPEEIGTTLLAPSLA\"\n",
      "Reading 188958 characters from file ./results_lopho//sim_match_lophotroch/DB/SE021.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE021.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE021.db.tree with 83224 entries\n",
      "Pat index with 83224 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYQPPLAAEGLNKIRASH\"\n",
      "Reading 191020 characters from file ./results_lopho//sim_match_lophotroch/DB/SE022.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE022.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE022.db.tree with 83516 entries\n",
      "Pat index with 83516 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTLLLLGLLYVERSDAI\"\n",
      "Reading 195311 characters from file ./results_lopho//sim_match_lophotroch/DB/SE023.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE023.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE023.db.tree with 83027 entries\n",
      "Pat index with 83027 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSWVNSNRRYTTSTPAY\"\n",
      "Reading 183251 characters from file ./results_lopho//sim_match_lophotroch/DB/SE024.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE024.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE024.db.tree with 81983 entries\n",
      "Pat index with 81983 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWLDIMDYLSKSKDNGI\"\n",
      "Reading 199565 characters from file ./results_lopho//sim_match_lophotroch/DB/SE025.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE025.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE025.db.tree with 83252 entries\n",
      "Pat index with 83252 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTSSKWGVGKSHRIYFL\"\n",
      "Reading 186561 characters from file ./results_lopho//sim_match_lophotroch/DB/SE026.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE026.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE026.db.tree with 83306 entries\n",
      "Pat index with 83306 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWIQKKLQERIALLRNV\"\n",
      "Reading 184428 characters from file ./results_lopho//sim_match_lophotroch/DB/SE027.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE027.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE027.db.tree with 82233 entries\n",
      "Pat index with 82233 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWPQPRIAEDMLLFSDA\"\n",
      "Reading 201732 characters from file ./results_lopho//sim_match_lophotroch/DB/SE028.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE028.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE028.db.tree with 83198 entries\n",
      "Pat index with 83198 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYWLAKDPLIFTPVGEGME\"\n",
      "Reading 200438 characters from file ./results_lopho//sim_match_lophotroch/DB/SE029.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE029.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE029.db.tree with 81447 entries\n",
      "Pat index with 81447 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYRPTLLSQVAHGKLFH\"\n",
      "Reading 191710 characters from file ./results_lopho//sim_match_lophotroch/DB/SE030.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE030.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE030.db.tree with 83119 entries\n",
      "Pat index with 83119 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYWRPLTNINGELGGENDG\"\n",
      "Reading 187093 characters from file ./results_lopho//sim_match_lophotroch/DB/SE031.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE031.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE031.db.tree with 82644 entries\n",
      "Pat index with 82644 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTVKDHIVFIPGQLNFL\"\n",
      "Reading 181775 characters from file ./results_lopho//sim_match_lophotroch/DB/SE032.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE032.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE032.db.tree with 82728 entries\n",
      "Pat index with 82728 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTEVVTRTMVSRDQTMQ\"\n",
      "Reading 216436 characters from file ./results_lopho//sim_match_lophotroch/DB/SE033.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE033.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE033.db.tree with 82804 entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pat index with 82804 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYKPIALPTPTVMLLDCK\"\n",
      "Reading 191367 characters from file ./results_lopho//sim_match_lophotroch/DB/SE034.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE034.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE034.db.tree with 82876 entries\n",
      "Pat index with 82876 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTAPGFKLNNFTYKKEL\"\n",
      "Reading 203267 characters from file ./results_lopho//sim_match_lophotroch/DB/SE035.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE035.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE035.db.tree with 81727 entries\n",
      "Pat index with 81727 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYRCTKVFTS</SEQ></\"\n",
      "Reading 199737 characters from file ./results_lopho//sim_match_lophotroch/DB/SE036.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE036.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE036.db.tree with 83157 entries\n",
      "Pat index with 83157 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVLYRSTLFVAACADIH\"\n",
      "Reading 203532 characters from file ./results_lopho//sim_match_lophotroch/DB/SE037.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE037.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE037.db.tree with 82533 entries\n",
      "Pat index with 82533 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSADNLLIIEAGVARWY\"\n",
      "Reading 196288 characters from file ./results_lopho//sim_match_lophotroch/DB/SE038.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE038.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE038.db.tree with 83072 entries\n",
      "Pat index with 83072 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSTMVSTPEPNEEKPKK\"\n",
      "Reading 202365 characters from file ./results_lopho//sim_match_lophotroch/DB/SE039.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE039.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE039.db.tree with 83089 entries\n",
      "Pat index with 83089 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYFEVIITQERLAFRALK\"\n",
      "Reading 209433 characters from file ./results_lopho//sim_match_lophotroch/DB/SE040.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE040.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE040.db.tree with 82309 entries\n",
      "Pat index with 82309 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYMIGTARISKKSIVCFY\"\n",
      "Reading 192023 characters from file ./results_lopho//sim_match_lophotroch/DB/SE041.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE041.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE041.db.tree with 83172 entries\n",
      "Pat index with 83172 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYLYQTTLMSVATGDAT\"\n",
      "Reading 196520 characters from file ./results_lopho//sim_match_lophotroch/DB/SE042.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE042.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE042.db.tree with 82944 entries\n",
      "Pat index with 82944 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYQMPCATKEMKTAFPA\"\n",
      "Reading 188897 characters from file ./results_lopho//sim_match_lophotroch/DB/SE043.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE043.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE043.db.tree with 82598 entries\n",
      "Pat index with 82598 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVVAQACGHEGAPERTR\"\n",
      "Reading 197093 characters from file ./results_lopho//sim_match_lophotroch/DB/SE044.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE044.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE044.db.tree with 83164 entries\n",
      "Pat index with 83164 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWVQKELQAKIATLHDN\"\n",
      "Reading 201027 characters from file ./results_lopho//sim_match_lophotroch/DB/SE045.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE045.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE045.db.tree with 83058 entries\n",
      "Pat index with 83058 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSWVFSNRRYTTSTPAY\"\n",
      "Reading 203405 characters from file ./results_lopho//sim_match_lophotroch/DB/SE046.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE046.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE046.db.tree with 83040 entries\n",
      "Pat index with 83040 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYEKGGENLFQEIGHVLR\"\n",
      "Reading 201931 characters from file ./results_lopho//sim_match_lophotroch/DB/SE047.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE047.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE047.db.tree with 83046 entries\n",
      "Pat index with 83046 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWIPKELQAKVGMLHNN\"\n",
      "Reading 195420 characters from file ./results_lopho//sim_match_lophotroch/DB/SE048.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE048.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE048.db.tree with 83068 entries\n",
      "Pat index with 83068 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYQSARLGSPLPKMLSDP\"\n",
      "Reading 193064 characters from file ./results_lopho//sim_match_lophotroch/DB/SE049.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE049.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE049.db.tree with 82850 entries\n",
      "Pat index with 82850 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYGEKMFNREGNEHTSK\"\n",
      "Reading 201149 characters from file ./results_lopho//sim_match_lophotroch/DB/SE050.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE050.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE050.db.tree with 83038 entries\n",
      "Pat index with 83038 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYQSARLGSPLPKMLSDP\"\n",
      "Reading 200864 characters from file ./results_lopho//sim_match_lophotroch/DB/SE051.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE051.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE051.db.tree with 83028 entries\n",
      "Pat index with 83028 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYQSARLGSPLPKMLSDP\"\n",
      "Reading 200596 characters from file ./results_lopho//sim_match_lophotroch/DB/SE052.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE052.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE052.db.tree with 82632 entries\n",
      "Pat index with 82632 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLGIYIEEGKKVLTGAQ\"\n",
      "Reading 208020 characters from file ./results_lopho//sim_match_lophotroch/DB/SE053.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE053.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE053.db.tree with 83028 entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pat index with 83028 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYIDEGLLLQTYGAPKR\"\n",
      "Reading 214093 characters from file ./results_lopho//sim_match_lophotroch/DB/SE054.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE054.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE054.db.tree with 82173 entries\n",
      "Pat index with 82173 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLGHDGSQTSPKSFARW\"\n",
      "Reading 215993 characters from file ./results_lopho//sim_match_lophotroch/DB/SE055.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE055.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE055.db.tree with 82235 entries\n",
      "Pat index with 82235 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLDHDGSQTSPKSFARW\"\n",
      "Reading 203972 characters from file ./results_lopho//sim_match_lophotroch/DB/SE056.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE056.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE056.db.tree with 83270 entries\n",
      "Pat index with 83270 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSGPEDIGTTLLTPALA\"\n",
      "Reading 212128 characters from file ./results_lopho//sim_match_lophotroch/DB/SE057.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE057.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE057.db.tree with 82425 entries\n",
      "Pat index with 82425 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSRDSLLIIEAGVAQWY\"\n",
      "Reading 227332 characters from file ./results_lopho//sim_match_lophotroch/DB/SE058.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE058.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE058.db.tree with 82846 entries\n",
      "Pat index with 82846 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVEIDATQEEQSSFITC\"\n",
      "Reading 229677 characters from file ./results_lopho//sim_match_lophotroch/DB/SE059.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE059.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE059.db.tree with 83249 entries\n",
      "Pat index with 83249 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVLKNCEDSYFHAGEYA\"\n",
      "Reading 236677 characters from file ./results_lopho//sim_match_lophotroch/DB/SE060.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE060.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE060.db.tree with 82551 entries\n",
      "Pat index with 82551 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWDKPVLLFSDDRLFRF\"\n",
      "Reading 205776 characters from file ./results_lopho//sim_match_lophotroch/DB/SE061.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE061.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE061.db.tree with 83237 entries\n",
      "Pat index with 83237 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSRYYSSEDVLSADPQN\"\n",
      "Reading 240379 characters from file ./results_lopho//sim_match_lophotroch/DB/SE062.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_lopho//sim_match_lophotroch/DB/SE062.db.map for mapping\n",
      "Building new Pat index in file ./results_lopho//sim_match_lophotroch/DB/SE062.db.tree with 83054 entries\n",
      "Pat index with 83054 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLFDEPGPLTPVSSVQD\"\n",
      "\n",
      "\n",
      ".fasta files written...\n",
      "\n",
      "\n",
      "...genome DBs saved.\n",
      "\n",
      "   simulation finished!\n",
      "\n",
      "\n",
      "synthetic evolution completed in 1.11 minutes!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!alfsim alf-params_lopho.drw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: /usr/local/bin/../lib/simulator/splashscreen.txt: No such file or directory\n",
      "Darwin: Sequence Searching Facility\n",
      "Version 4.0, 2014-02-21\n",
      "  (c) E.T.H. Zurich\n",
      "> paramfile := 'alf-params_myria.drw': wdir := './': ReadProgram(libname.'/simulator/evolstart.drw'): done:\n",
      "\n",
      "IO procedures for Synthetic Evolution sucessfully loaded\n",
      "Procedures for Genome Rearrangement sucessfully loaded\n",
      "Procedures for Synthetic Evolution sucessfully loaded\n",
      "\n",
      "\n",
      "parameter file read\n",
      "setting up work directory\n",
      "determining tree\n",
      "\n",
      "tree loaded\n",
      "model assignments done\n",
      "\n",
      "\n",
      "Evolution!\n",
      "\n",
      "\n",
      "species 2 was created at time 0.0000\n",
      "species 3 was created at time 2.0217\n",
      "species 4 was created at time 44.8963\n",
      "species 5 was created at time 46.5048\n",
      "species 6 was created at time 70.1910\n",
      "species 7 was created at time 70.8892\n",
      "species 8 was created at time 70.9746\n",
      "species 9 was created at time 75.6986\n",
      "species 10 was created at time 77.6222\n",
      "species 11 was created at time 82.9531\n",
      "species 12 was created at time 86.9097\n",
      "species 13 was created at time 91.5299\n",
      "species 14 was created at time 95.6776\n",
      "species 15 was created at time 99.3204\n",
      "species 16 was created at time 100.7883\n",
      "species 17 was created at time 100.8293\n",
      "species 18 was created at time 107.9032\n",
      "species 19 was created at time 108.5222\n",
      "species 20 was created at time 109.4637\n",
      "species 21 was created at time 111.2429\n",
      "species 22 was created at time 142.4616\n",
      "species 23 was created at time 147.5029\n",
      "species 24 was created at time 148.5781\n",
      "species 25 was created at time 149.9060\n",
      "...real genome tree saved.\n",
      "...main calculation finished after 0.17 minutes.\n",
      "tar: Option --remove-files is not supported\n",
      "Usage:\n",
      "  List:    tar -tf <archive-filename>\n",
      "  Extract: tar -xf <archive-filename>\n",
      "  Create:  tar -cf <archive-filename> [filenames...]\n",
      "  Help:    tar --help\n",
      "...pairwise relationships written.\n",
      "...gene trees written.\n",
      "...real MSAs written.\n",
      "\n",
      "\n",
      ".db files written...\n",
      "\n",
      "\n",
      "Reading 157246 characters from file ./results_myria//sim_match_myriapod/DB/SE001.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE001.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE001.db.tree with 83159 entries\n",
      "Pat index with 83159 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYWTVRGRIKNPKDLNFAQ\"\n",
      "Reading 208070 characters from file ./results_myria//sim_match_myriapod/DB/SE002.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE002.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE002.db.tree with 83964 entries\n",
      "Pat index with 83964 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYRVATGEWELLRDCLHV\"\n",
      "Reading 164733 characters from file ./results_myria//sim_match_myriapod/DB/SE003.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE003.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE003.db.tree with 82665 entries\n",
      "Pat index with 82665 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTKTSEVLASAHKLKVF\"\n",
      "Reading 158519 characters from file ./results_myria//sim_match_myriapod/DB/SE004.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE004.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE004.db.tree with 82644 entries\n",
      "Pat index with 82644 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWEAPEFANEGFTEKTI\"\n",
      "Reading 157766 characters from file ./results_myria//sim_match_myriapod/DB/SE005.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE005.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE005.db.tree with 82890 entries\n",
      "Pat index with 82890 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLHQHNPALLDNAYGED\"\n",
      "Reading 166949 characters from file ./results_myria//sim_match_myriapod/DB/SE006.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE006.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE006.db.tree with 82745 entries\n",
      "Pat index with 82745 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWVGEADGETESFVADH\"\n",
      "Reading 187747 characters from file ./results_myria//sim_match_myriapod/DB/SE007.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE007.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE007.db.tree with 82746 entries\n",
      "Pat index with 82746 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSRRTSLGTHIPSLPLN\"\n",
      "Reading 157180 characters from file ./results_myria//sim_match_myriapod/DB/SE008.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE008.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE008.db.tree with 83133 entries\n",
      "Pat index with 83133 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYKALGRTISPGRLATLV\"\n",
      "Reading 170377 characters from file ./results_myria//sim_match_myriapod/DB/SE009.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE009.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE009.db.tree with 82697 entries\n",
      "Pat index with 82697 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYKMNVLLRLAIKSVVIH\"\n",
      "Reading 167695 characters from file ./results_myria//sim_match_myriapod/DB/SE010.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE010.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE010.db.tree with 82501 entries\n",
      "Pat index with 82501 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYPRHNTAFPTSFMCTTV\"\n",
      "Reading 166040 characters from file ./results_myria//sim_match_myriapod/DB/SE011.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE011.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE011.db.tree with 82623 entries\n",
      "Pat index with 82623 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVYVCEKSTDSYPVFDD\"\n",
      "Reading 164321 characters from file ./results_myria//sim_match_myriapod/DB/SE012.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE012.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE012.db.tree with 82932 entries\n",
      "Pat index with 82932 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYKALGKTMSPGRLASLL\"\n",
      "Reading 164739 characters from file ./results_myria//sim_match_myriapod/DB/SE013.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE013.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE013.db.tree with 82801 entries\n",
      "Pat index with 82801 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYNRRNLAFLHTRNYSNL\"\n",
      "Reading 165899 characters from file ./results_myria//sim_match_myriapod/DB/SE014.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE014.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE014.db.tree with 83231 entries\n",
      "Pat index with 83231 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVHGPILEWEFIWKGRH\"\n",
      "Reading 179123 characters from file ./results_myria//sim_match_myriapod/DB/SE015.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE015.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE015.db.tree with 82720 entries\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pat index with 82720 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYVREPIMREFVKSYEHT\"\n",
      "Reading 185263 characters from file ./results_myria//sim_match_myriapod/DB/SE016.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE016.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE016.db.tree with 83532 entries\n",
      "Pat index with 83532 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYYVSESQLKRKGPAPNV\"\n",
      "Reading 168760 characters from file ./results_myria//sim_match_myriapod/DB/SE017.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE017.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE017.db.tree with 83231 entries\n",
      "Pat index with 83231 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYNQVYEQTNAAIEEESD\"\n",
      "Reading 171081 characters from file ./results_myria//sim_match_myriapod/DB/SE018.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE018.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE018.db.tree with 82627 entries\n",
      "Pat index with 82627 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWVGEADGETEGFVASH\"\n",
      "Reading 180132 characters from file ./results_myria//sim_match_myriapod/DB/SE019.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE019.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE019.db.tree with 82765 entries\n",
      "Pat index with 82765 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYWVGEADGETESYVAGH\"\n",
      "Reading 171699 characters from file ./results_myria//sim_match_myriapod/DB/SE020.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE020.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE020.db.tree with 83234 entries\n",
      "Pat index with 83234 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYWKSNYMKSQGRMADINM\"\n",
      "Reading 175796 characters from file ./results_myria//sim_match_myriapod/DB/SE021.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE021.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE021.db.tree with 83230 entries\n",
      "Pat index with 83230 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYLLTEEKQNNLKLYSSA\"\n",
      "Reading 183221 characters from file ./results_myria//sim_match_myriapod/DB/SE022.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE022.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE022.db.tree with 82823 entries\n",
      "Pat index with 82823 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYPKLKRGEERFGSVTGI\"\n",
      "Reading 184172 characters from file ./results_myria//sim_match_myriapod/DB/SE023.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE023.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE023.db.tree with 82859 entries\n",
      "Pat index with 82859 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYTGVPYGDPRAQLILNR\"\n",
      "Reading 188401 characters from file ./results_myria//sim_match_myriapod/DB/SE024.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE024.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE024.db.tree with 82730 entries\n",
      "Pat index with 82730 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYIIESKATVGEPENDMA\"\n",
      "Reading 194915 characters from file ./results_myria//sim_match_myriapod/DB/SE025.db\n",
      "Pre-processing input (peptides)\n",
      "200 sequences within 200 entries considered\n",
      "Creating file ./results_myria//sim_match_myriapod/DB/SE025.db.map for mapping\n",
      "Building new Pat index in file ./results_myria//sim_match_myriapod/DB/SE025.db.tree with 82676 entries\n",
      "Pat index with 82676 entries\n",
      " sorted, from \"A</SEQ></E>\\n<E><ID>\" to \"YYYSKIRRGEERFGNVNGI\"\n",
      "\n",
      "\n",
      ".fasta files written...\n",
      "\n",
      "\n",
      "...genome DBs saved.\n",
      "\n",
      "   simulation finished!\n",
      "\n",
      "\n",
      "synthetic evolution completed in 0.30 minutes!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!alfsim alf-params_myria.drw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subsample the output MSAs\n",
    "\n",
    "- full genomes (461 amino acids currently ?)\n",
    "- half the genome\n",
    "- 1/4\n",
    "- 1/10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write a script to loop through each file and make all the subsampled files\n",
    "# put those subsampled files (labelled properly by spp and subsamp size) \n",
    "#    into a new folder to more easily do analyses on them\n",
    "\n",
    "!./SampleMSAs.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Partition (or don't) the subampled (or full) output MSAs from the simulations\n",
    "\n",
    "- run PartitionFinder2\n",
    "- needs anaconda bc it runs on an old version of Python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda2/envs/py3/lib/python3.7/site-packages/ipykernel_launcher.py:9: DeprecationWarning: 'U' mode is deprecated\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "# convert .fa to .phylip for partition finder 2\n",
    "\n",
    "import glob, os\n",
    "from Bio import AlignIO\n",
    "\n",
    "FOLDER = 'sampledMSAs'\n",
    "\n",
    "for f in glob.glob(os.path.join(FOLDER, '*.fa')):\n",
    "    input_handle = open(f, \"rU\")\n",
    "    output_handle = open(f.replace('fa','phy'), \"w\")\n",
    "\n",
    "    align = AlignIO.read(input_handle, \"fasta\")\n",
    "    AlignIO.write(align, output_handle, \"phylip-relaxed\")\n",
    "\n",
    "    output_handle.close()\n",
    "    input_handle.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use a script I set up to go in the command line in the anaconda environment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Build the trees from the partitioned (or not partitioned) results\n",
    "\n",
    "- run IQtree\n",
    "- run RaxML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# can use the zoo wrappers for this step\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Compare the original simulated (true) trees to the inferred trees that were partitioned or not\n",
    "\n",
    "- use Robinson Foulds distance\n",
    "- use Euclidean distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Jeremy's RF calculation script\n",
    "\"\"\"\n",
    "Calculates RF distances between two trees\n",
    "\"\"\"\n",
    "\n",
    "from ete3 import Tree\n",
    "\n",
    "def collapse_branches(TREE_FILE,SUPPORT_THRESHOLD):\n",
    "    t = Tree(TREE_FILE)\n",
    "    for node in t.get_descendants():\n",
    "        if not node.is_leaf() and (node.support <= SUPPORT_THRESHOLD):\n",
    "            node.delete()\n",
    "    return t\n",
    "\n",
    "\n",
    "def count_internal(tree):\n",
    "    tree.unroot()\n",
    "    edges=-1\n",
    "    for edge in tree.traverse():\n",
    "        if not edge.is_leaf():\n",
    "            edges+=1\n",
    "    return edges\n",
    "\n",
    "def rf_distance(tree1,tree2,option=False):\n",
    "    if option=='collapse':\n",
    "        t1 = collapse_branches(tree1,0.75)\n",
    "        t2 = collapse_branches(tree2,0.75)\n",
    "        option = 'reduced'\n",
    "    else:\n",
    "        t1 = Tree(tree1)\n",
    "        t2 = Tree(tree2)\n",
    "\n",
    "    t1.unroot()\n",
    "    t2.unroot()\n",
    "\n",
    "    rf = t1.robinson_foulds(t2,unrooted_trees=True)\n",
    "\n",
    "    rf_dist = rf[0]\n",
    "    max_rf = rf[1]\n",
    "    num_leaves=len(rf[2])\n",
    "\n",
    "    max_resolved_score = (2*num_leaves)-6\n",
    "\n",
    "    internal_branches_t1 = count_internal(t1)\n",
    "    internal_branches_t2 = count_internal(t2)\n",
    "\n",
    "    total_internal = internal_branches_t1 + internal_branches_t2\n",
    "\n",
    "    num_missing_splits_t1 = num_leaves - 3 - internal_branches_t1\n",
    "    num_missing_splits_t2 = num_leaves - 3 - internal_branches_t2\n",
    "\n",
    "\n",
    "\n",
    "    rf_dist_upper = rf_dist + num_missing_splits_t1 + num_missing_splits_t2\n",
    "\n",
    "\n",
    "    #If normalise by (num intenral branches)\n",
    "    if option=='reduced':\n",
    "        normalised_rf = rf_dist/total_internal\n",
    "\n",
    "    elif option=='upper':\n",
    "    #If add score to create upper bound due to being polytomy\n",
    "        normalised_rf = rf_dist_upper/max_resolved_score\n",
    "    else:\n",
    "        normalised_rf = rf_dist/max_resolved_score\n",
    "\n",
    "    return normalised_rf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# compare the trees"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If there is overfitting, we expect the prtitioned trees to be no better matched to the true tree, or potentially even more poorly matched?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
